linera-views/src/memory.rs

```diff
@@ -47,12 +48,16 @@ pub const TEST_MEMORY_MAX_STREAM_QUERIES: usize = 10;
 /// The analog of the database is the BTreeMap
 type MemoryStoreMap = BTreeMap<Vec<u8>, Vec<u8>>;
 
-/// The container for the `MemoryStopMap` according to the Namespace.
-type NamespaceMemoryStore = BTreeMap<String, Arc<RwLock<MemoryStoreMap>>>;
+/// The container for the `MemoryStopMap` according to the Namespace and Namespace/root_key
+#[derive(Default)]
+struct MemoryStores {
+    stores: BTreeMap<(String, Vec<u8>), Arc<RwLock<MemoryStoreMap>>>,
+    namespaces: BTreeSet<String>,
```

Comment:
Would two nested maps be better? At least then it's impossible to represent a `stores` entry without a matching `namespaces` entry. And a namespace can still be empty (contain an empty `BTreeMap`).

```suggestion
    stores: BTreeMap<String, BTreeMap<Vec<u8>, Arc<RwLock<MemoryStoreMap>>>>,
```

Reply:
You are right, that led to simplification of the code.